### ISSUE TITLE
Media playback problem

### ISSUE THREAD

**Issue by northpole-pl** [Cimbali/pympress#197]
### Environment
- Windows 7
- Python version 3.8.8

### Pympress
- version 1.6.1
- Installation method: binary installer

### Problem
During startup, pympress reports errors:

```
[000000000603b280] mmdevice audio output error: cannot initialize COM (error 0x80010106)
[000000000603b280] mmdevice audio output error: cannot initialize COM (error 0x80010106)
```

Pympress.log:

```
INFO:pympress.app:Pympress: 1.6.1 ; Python: 3.8.9 ; OS: Windows 7 6.1.7601 ; Gtk 3.24.29 ; GLib 2.68.1 ; Poppler 20.12.1 cairo ; Cairo 1.17.4 , pycairo 1.20.0
DEBUG:pympress.extras:Media support using GStreamer is disabled.
DEBUG:pympress.extras:Caused by ValueError: Namespace Gst not available
INFO:pympress.extras:Media support: GdkPixbuf gif player, GStreamer not available, VLC 3.0.14 Vetinari
WARNING:pympress.ui:Not starting content or presenter window full screen because there is only one monitor
ERROR:pympress.scribble:Error creating highlight cache
Traceback (most recent call last):
  File "D:\a\pympress\pympress/pympress/scribble.py", line 342, in reset_scribble_cache
  File "D:/a/_temp/msys/msys64/mingw64/lib/python3.8/site-packages/gi/module.py", line 155, in __getattr__
ValueError: invalid enum value: 5
```

pympress does not play media files (mp4, avi, gif)

**Comment by Cimbali** [comment]
So nothing unusual in the logs (the only error there is expected and properly handled), and VLC should be enabled. VLC’s COM error message seems to be the same as [this one reported on the VLC forums](https://forum.videolan.org/viewtopic.php?t=146741):


> That error is non fatal. LibVLC will fallback to DirectSound - and it really does not matter if you disable audio.
>
> -- <cite>Postby Rémi Denis-Courmont » [20 Oct 2018 12:18](https://forum.videolan.org/viewtopic.php?p=481440&sid=79e1328bde5f38e3405812e9bd688151#p481440)</cite>

I suspect a regression in the code that selects which backend to use. Not sure why it only seems to happen on Windows though. Thanks for reporting anyway, I’ll look into it as soon as I have some time !

**Comment by llimeht** [comment]
Some experimentation with this problem makes it look like a set of filename handling issues:

pympress has a different idea of what the PDF filename is depending on how the file is loaded. 
- if the filename is passed in from the command line or from the file manager, `swap_document` is given a URL that looks like `file:///C:/…` and `Document.__init__` gets `/C:/Users/.../foo.pdf`; later attempts to find an external media file tries to construct filenames that don't exist e.g. `/C:/Users/…/video.mp4` so there is not even an attempt to load the media. `get_annot_action()` realises that there is no media to be found as `self.parent.get_full_path` falls through to `None`. The log file contains the expected "can not find file /C:/Users/…/video.mp4".
- alternatively, if the PDF is opened from File → Open, `swap_document` is given a file path `C:/Users/…/foo.pdf` and the media is found.

I see in the code that there are other places where the number of `/` on windows has been an issue so this is presumably a vexing issue.

I did find that just re-opening the file in pympress would let me watch the video, but I also sometimes ended up with the video popped-out rather than overlayed, so I'm unconvinced that I got it working correctly.

I'm happy to investigate further but would need some suggestion of what to look at next; windows is not my primary platform and I have little experience with file path handling in Python on this platform.
